### DISCUSSION TITLE
Create JUMP_TO instruction on LLIL

### DISCUSSION THREAD

**Discussion by NeoZ16** [Vector35/binaryninja-api#4113] [API] [answered]
I have an indirect jump to a register, like `jmp rax` and want to create a JUMP_TO expression with my own possible targets.
From what I've gathered so far, you need to create a new expression with `function.llil.expr(LLIL_JUMP_TO)`.
But the parameters have the type ExpressionIndex.

I've found this `LowLevelILOperation.LLIL_JUMP_TO: LowLevelILJumpTo, #  [("dest", "expr"), ("targets", "target_map")]`, but I can't really make sense of it.

I'm not sure on how to create an ExpressionIndex from a register of if I have to pass something else and I was not able to find out how to pass the `target_map`.

Maybe I'm missing something really simple here or there is an example that explains this, but I can't seem to find it.

**Comment by rssor** [comment, marked as answer]
It sounds like what you want is `Function.set_user_indirect_branches`.

**Reply by psifertex** [reply]
An example showing that API is here: https://github.com/Vector35/binaryninja-api/blob/dev/python/examples/jump_table.py
